src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -670,9 +673,14 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                         throw ExceptionUtilities.UnexpectedValue(_deconstruction.Kind());
                 }
 
-                TypeSymbol result = this._type;
-                Debug.Assert((object)result != null);
-                return result;
+                if ((object)this._type == null)
+                {
+                    diagnostics.Add(this.ForbiddenDiagnostic, Locations[0], Name);
+                    SetType(new ExtendedErrorTypeSymbol(
+                        ((SourceModuleSymbol)this.ContainingModule).DeclaringCompilation, name: "var", arity: 0, errorInfo: null, variableUsedBeforeDeclaration: true));
```

Comment:
It wouldn't have any effect other than recording that the variable's type was required at a location before it was able to be inferred.

---
In reply to: [91395381](https://github.com/dotnet/roslyn/pull/15548#discussion_r91395381) [](ancestors = 91395381)